### ISSUE TITLE
Delegate trajectory properties to sample

### ISSUE THREAD

**Issue by jhprinz** [choderalab/openpathsampling#257]
I realized that we often use a sample as a trajectory since it basically is a trajectory with additional
information (although samples can have empty trajectories or even None).

This would mostly allow to write
```
len(sample) # length of the trajectory
sample[slice] # same as sample.trajectory[slice]
```
in addition, use samples in order parameters and ensembles
```
ensemble(sample) # is ensemble(sample.trajectory)
# note that 
sample.valid = sample.ensemble(sample.trajectory)
cv1 = CollectiveVariable()
cv1(sample) = cv1(sample.trajectory)
```

I think this makes scripts more readable and shorter. In the long run, we might be able to use trajectories only internally with no need to expose these. Like we do with configuration or momentum and also with MCStep

**Comment by dwhswenson** [comment]
I think this is a very good idea. I think the main usefulness would be that we could, in many cases, directly use:

```python
function_that_takes_trajectory_argument(sample)
```

I can't count the times I've tried to use a sample as a trajectory when showing live code to Peter. If *I'm* doing it (given that I should know better), then that's probably the behavior a user would expect.

Note that, in one interpretation, this means that `Sample` becomes a `Trajectory` with extra metadata. I don't think that version should be reflected in the implementation, but it might be worth trying write code that supports that interpretation.

**Comment by dwhswenson** [comment]
Closed by #265.
